homeassistant/components/climate/generic_thermostat.py

```diff
@@ -166,7 +166,8 @@ def async_added_to_hass(self):
                 else:
                     self._target_temp = float(
                         old_state.attributes[ATTR_TEMPERATURE])
-            if old_state.attributes[ATTR_AWAY_MODE] is not None:
+            if ATTR_AWAY_MODE in old_state.attributes \
+                    and old_state.attributes[ATTR_AWAY_MODE] is not None:
```

Comment:
`if old_state.attributes.get(ATTR_AWAY_MODE) is not None:`
Is maybe better?

Reply:
And so I improve my python knowledge every day :-)